server/src/test/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateServiceTests.java

```diff
@@ -859,6 +919,119 @@ public void testRemoveComponentTemplateInUse() throws Exception {
             containsString("component templates [ct] cannot be removed as they are still in use by index templates [template]"));
     }
 
+    /**
+     * Tests that we check that settings/mappings/etc are valid even after template composition,
+     * when adding/updating a composable index template
+     */
+    public void testInvalidComposableCompositeTemplate() throws Exception {
```

Comment:
this and the next test names are rather confusing to me. It's quite difficult to pick up what they are testing and what's the invalid case. (maybe related to the previous comment too, but  "composable composite" adds to the confusion for me)

what do you think of including the cause of failure in the test name? 

What do you think of something along the lines of  `testIndexTemplateFailsToOverrideComponentTemplateMappingField` and respectively `testUpdateComponentTemplateFailsIfResolvedIndexTemplatesWouldBeInvalid` ?